pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -492,35 +504,39 @@ func (c *Controller) syncCheckStates(ctx context.Context, wl *kueue.Workload, ch
 
 			prFailed := apimeta.IsStatusConditionTrue(pr.Status.Conditions, autoscaling.Failed)
 			prProvisioned := apimeta.IsStatusConditionTrue(pr.Status.Conditions, autoscaling.Provisioned)
-			log.V(3).Info("Synchronizing admission check state based on provisioning request", "wl", klog.KObj(wl), "check", check, "prName", pr.Name, "failed", prFailed, "accepted", prProvisioned)
+			prAccepted := apimeta.IsStatusConditionTrue(pr.Status.Conditions, autoscaling.Accepted)
+			log.V(3).Info("Synchronizing admission check state based on provisioning request", "wl", klog.KObj(wl),
+				"check", check,
+				"prName", pr.Name,
+				"failed", prFailed,
+				"provisioned", prProvisioned,
+				"accepted", prAccepted)
 
 			switch {
 			case prFailed:
 				if checkState.State != kueue.CheckStateRejected {
 					if attempt := getAttempt(ctx, pr, wl.Name, check); attempt <= MaxRetries {
 						// it is going to be retried
 						message := fmt.Sprintf("Retrying after failure: %s", apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Failed).Message)
-						updated = updated || checkState.State != kueue.CheckStatePending || checkState.Message != message
-						checkState.State = kueue.CheckStatePending
-						checkState.Message = message
+						updated = updateCheckState(&checkState, kueue.CheckStatePending) || updated
+						updated = updateCheckMessage(&checkState, message) || updated
 					} else {
 						updated = true
 						checkState.State = kueue.CheckStateRejected
 						checkState.Message = apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Failed).Message
 					}
 				}
 			case prProvisioned:
-				if checkState.State != kueue.CheckStateReady {
+				if updateCheckState(&checkState, kueue.CheckStateReady) {
 					updated = true
-					checkState.State = kueue.CheckStateReady
 					// add the pod podSetUpdates
 					checkState.PodSetUpdates = podSetUpdates(wl, pr)
 				}
 			default:
-				if checkState.State != kueue.CheckStatePending {
-					updated = true
-					checkState.State = kueue.CheckStatePending
-				}
+				updated = updateCheckState(&checkState, kueue.CheckStatePending) || updated
+			}
+			if prAccepted && !prFailed {
+				updated = updateCheckMessage(&checkState, apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Provisioned).Message) || updated
```

Comment:
maybe add a case for `prAccepted` in the `switch` so you can do the message update there?

Reply:
yeah, it seems like it would fit into the `switch` statement. But adding a case for `prAccepted` would cause the `default` case not to be executed and this is not what we want (bc we update a state there). As I understand it, the switch is mainly for updating the state of the admission check. Since my feature will be about only updating the message I don't want to mess with the `switch` statement. I did consider it, thought about using  `fallthrough`, but I could not find a good way that would not make this code more brittle.

Reply:
Could you elaborate why the switch is mainly for updating the state of admission check? I imagine it could be along the lines: 

```golang
switch {
 case prProvisioned: // updates state and message (ETA -> successfully provisioned)
 case prFailed: // retries or updates state and message (ETA -> failed)
 case prAccepted && !prFailed : // updates message (ETA)
 default: // for now covers cases !prAccepted, prBookingExpired, prCapacityRevoked
}
```

Reply:
I see. I wanted to avoid having the ETA update in multiple places, but perhaps you are right. One more thing is that in the new `prAccepted` case I would have to also run everything that is in the `default` case. But, I'll revisit this idea.

Reply:
Is `prAccepted` different case than `prAccepted && !prFailed`? I thought `prFailed` is by default set to `false`

Reply:
I might be missing something, but switch only runs the single case; the topmost one that evaluates to true, right? so if we reach to `case prAccepted` we are sure that `prFailed = False` and `prProvisioned= False`, right?